sklearn/preprocessing/_encoders.py

```diff
@@ -467,6 +484,7 @@ def _validate_keywords(self):
                 f"'infrequent_if_exist' got {self.handle_unknown}."
             )
             raise ValueError(msg)
+        self._get_feature_name_combiner()  # call to force the validation
```

Comment:
Since we are also using the function for validation:

```suggestion
        self._check_get_feature_name_combiner()  # call to force the validation
```

WDYT @glemaitre ?